src/cas/flatten_data_to_anndata.py

```diff
@@ -214,3 +233,199 @@ def collect_parent_cell_ids(cas):
+
+            # Process each column in the group
+            for col in df.columns:
+                if col == df_key:
+                    # Store the labelset and cell label for the first column
+                    nested_dict[key][LABELSET] = col
+                    nested_dict[key][CELL_LABEL] = group[col].iloc[0]
+                    # Store cellhash
+                    cell_hash = accession_manager.generate_accession_id(
+                        cell_ids=cell_id_list, labelset=col
+                    )
+                    nested_dict[key][AUTHOR_ANNOTATION_FIELDS][CELLHASH] = cell_hash
+                else:
+                    # Split annotation columns and store them in annotations
+                    annotation_column = col.split("--")[-1]
+                    if annotation_column == AUTHOR_ANNOTATION_FIELDS:
+                        annotation_dict = json.loads(
+                            group[col].iloc[0].replace("'", '"')
+                        )
+                        filtered_annotation_dict = {
+                            k: v for k, v in annotation_dict.items() if k != CELLHASH
+                        }
+                        nested_dict[key][annotation_column].update(
+                            filtered_annotation_dict
+                        )
+                        continue
+                    nested_dict[key][annotation_column] = group[col].iloc[0]
+
+    return {
+        key_: dict(value)
+        for key, value in nested_dict.items()
+        for key_ in (key, value[AUTHOR_ANNOTATION_FIELDS][CELLHASH])
+    }
+
+
+def update_cas_json(
+    cas_dict: Dict[str, Dict[str, Any]], cas_json: Dict[str, Any]
+) -> Dict[str, Any]:
+    """
+    Update the annotations in the CAS JSON using the provided lookup dictionary.
+
+    This function checks the CAS JSON annotations against a lookup dictionary. It updates
+    annotations where cell labels or cell hashes match and discards mismatches. It also
+    adds new annotations from the lookup dictionary that are not in the CAS JSON.
+
+    Args:
+        cas_dict: A lookup dictionary where keys are cell labels or hashes, and
+                  values are dictionaries with annotation data.
+        cas_json: The CAS JSON object containing existing annotations.
+
+    Returns:
+        The updated CAS JSON with synchronized annotations based on the lookup dictionary.
+    """
+    updated_cas_annotations: List[Dict[str, Any]] = []
+    remaining_annotations_labels = list(key for key in cas_dict.keys())
+    # remaining_annotations_labels = list(key for key in cas_dict.keys() if not is_hash_accession(key))
+
+    for annotation in cas_json[ANNOTATIONS]:
+        cas_cell_label = annotation[CELL_LABEL]
+        cas_cellhash = annotation[AUTHOR_ANNOTATION_FIELDS][CELLHASH]
+
+        if cas_cell_label in cas_dict.keys():
+            obs_annotation = cas_dict[cas_cell_label]
+            obs_cellhash = cas_dict[cas_cell_label][AUTHOR_ANNOTATION_FIELDS][CELLHASH]
```

Comment:
Instead of only `cell_label` based matching, we need a `labelset:cell_label pair` based match according to the issue.